app/controllers/api/v1/invoices_controller.rb

```diff
@@ -31,6 +31,11 @@ def index
         if params[:external_customer_id]
           invoices = invoices.joins(:customer).where(customers: { external_id: params[:external_customer_id] })
         end
+
+        if valid_payment_status?(params[:payment_status])
```

Comment:
Wondering if all this logic should be part of a query class:

```ruby
class BaseQuery < BaseService
  def initialize(organization)
    @organization = organization

    super
  end

  private

  attr_reader :organization
end

class InvoiceQuery < BaseQuery
  def call(filters)
    invoices = base_scope.xxx

    result.invoices = invoices
    result
  end

  private

  def base_scope
    organization.invoices
  end
end
```

That way you can even handle error with filters, or have shared methods for pagination

In the controller:

```ruby
def index
  query = InvoiceQuery.new(current_organization)
  result = query.call(params.permit(:external_customer_id, :payment_status))

  if result.success?
    render(
        json: ::CollectionSerializer.new(
          invoices,
          ::V1::InvoiceSerializer,
          collection_name: 'invoices',
          meta: pagination_metadata(invoices),
        ),
      )
  else
    render_error_response(result)
  end
end
```

Reply:
opened a card for it :)